### PR TITLE
WIP: ENH: Add a more informative default naming convention

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: ibis/expr/tests/test_table.py [hunk 1]
```diff
@@ -693,27 +692,29 @@ def test_value_counts_unnamed_expr(con):
...
-        (d.approx_median(), 'approx_median'),
-        (d.min(), 'min'),
-        (d.max(), 'max')
+        (d.count(), 'count_f'),
+        (d.sum(), 'sum_f'),
+        (d.mean(), 'mean_f'),
+        (d.approx_nunique(), 'approxnunique_f'),
+        (d.approx_median(), 'approxmedian_f'),
+        (d.min(), 'min_f'),
+        (d.max(), 'max_f'),
```
Comment (wesm) [inline ibis/expr/tests/test_table.py]: Would parentheses be more like databases? No tab completion, though

File: ibis/sql/tests/test_compiler.py [hunk 2]
```diff
@@ -1305,15 +1305,16 @@ def test_bug_project_multiple_times(self):
   LEFT SEMI JOIN (
     SELECT *
     FROM (
-      SELECT t1.`n_name`, sum(CAST(t0.`c_acctbal` AS double)) AS `sum`
+      SELECT t1.`n_name`,
+             sum(CAST(t0.`c_acctbal` AS double)) AS `sum_cast(c_acctbal, double)`
```
Comment (wesm) [inline ibis/sql/tests/test_compiler.py]: hm, unsure about parens versus underscores

**Comment by cpcloud** [comment]
I dont think this will make it into 0.11 but could be in 0.11.1 and will definitely make 0.12.

**Comment by cpcloud** [comment]
Closing for now, will revisit after 0.12 is released.

**Pull request closed without merging**